src/components/NavigationTabBar.js

```diff
@@ -61,67 +61,44 @@ class NavigationTabBar extends React.PureComponent<_TabBarBottomProps> {
   };
 
   renderLabel = (scene: TabScene) => {
-    const { position, navigation } = this.props;
-    const { index } = scene;
-    const { routes } = navigation.state;
-
-    // Prepend '-1', so there are always at least 2 items in inputRange
-    const inputRange = [-1, ...routes.map((x, i) => i)];
-    const color = position.interpolate({
-      inputRange,
-      outputRange: inputRange.map(
-        inputIndex =>
-          inputIndex === index ? tabBarActiveLabelColor : tabBarLabelColor
-      )
-    });
-    const tintColor = scene.focused ? tabBarActiveLabelColor : tabBarLabelColor;
-    const label = this.props.getLabel({ ...scene, tintColor });
+    const { focused } = scene;
+    const color = focused ? tabBarActiveLabelColor : tabBarLabelColor;
+    const label = this.props.getLabelText(scene);
 
     return (
-      <Animated.Text
+      <Text
         numberOfLines={1}
         style={[styles.label, { color }]}
         allowFontScaling
       >
         {label}
-      </Animated.Text>
+      </Text>
     );
   };
 
   renderIcon = (scene: TabScene) => {
-    const { position, navigation } = this.props;
-    const { route, index } = scene;
-    const { routes } = navigation.state;
-
-    // Prepend '-1', so there are always at least 2 items in inputRange
-    const inputRange = [-1, ...routes.map((x, i) => i)];
-    const activeOpacity = position.interpolate({
-      inputRange,
-      outputRange: inputRange.map(i => (i === index ? 1 : 0))
-    });
-    const inactiveOpacity = position.interpolate({
-      inputRange,
-      outputRange: inputRange.map(i => (i === index ? 0 : 1))
-    });
+    const { route, index, focused } = scene;
+    const opacity = focused ? 1 : 0;
+    const opacitInv = focused ? 0 : 1;
```

Comment:
Oh nooooo, the animated `position` doesn't exist anymore? I think I was able to see/feel this animation of the opacity 😊.

Reply:
Yeah it's gone. We can roll our own though. We just could not see the difference so took it out.

Reply:
I've sorted the `tabBarVisible` issue, would love to hear if you have any ideas as to cleaner alternatives to my solution :)